pkg/controller/appstatus/endpoints.go

```diff
@@ -90,7 +94,17 @@ func serviceEndpoints(ctx context.Context, c kclient.Client, app *v1.AppInstance
 		}
 	}
 
-	return
+	set := map[string]v1.Endpoint{}
+	for _, ep := range endpoints {
+		set[fmt.Sprintf("%v://%v", ep.Protocol, ep.Address)] = ep
+	}
+
+	var result []v1.Endpoint
```

Comment:
nit to avoid multiple allocations:
```suggestion
	result := make([]v1.Endpoint, 0, len(set))
```